irods/test/data_obj_test.py

```diff
@@ -654,6 +675,168 @@ def test_multiple_reads(self):
+                                           ):
+        ticket = [None]
+        data = [None]
+        if '/' in data_name:
+            data_path = data_name
+        else:
+            if not data_name:
+                data_name = helpers.unique_name(helpers.my_function_name(), datetime.now())
+            data_path = helpers.home_collection(self.sess)+"/"+data_name
+        def create_data(open_options):
+            if content is not None and not self.sess.data_objects.exists(data_path):
+                with self.sess.data_objects.open(data_path,"w",**dict(open_options)) as f:
+                    f.write(content)
+            data[0] = data_path
+        try:
+            if ticket_access:
+                user_session = self.logins.session_for_user(RODSUSER)
+                def initialize(open_options=()):
+                    create_data(open_options)
+                    # Activate ticket for test session:
+                    ticket_logical_path = data_path if self.sess.data_objects.exists(data_path) else irods_dirname(data_path)
+                    ticket[0] = Ticket(self.sess).issue(ticket_access, ticket_logical_path)
+                    Ticket(user_session, ticket[0].string).supply()
+            else:
+                user_session = self.sess
+                initialize = lambda open_options=(): \
+                                 create_data(open_options)
+            # Can be called early to clear resources of data objects before their deletion
+            def finalize():
+                if auto_delete_data and data[0]:
+                    self.sess.data_objects.unlink(data[0], force=True)
+                    data[0]=None
+            yield { 'path':data_path, 'session':user_session, 'finalize': finalize, 'initialize':initialize, 'ticket_access': ticket_access }
+        finally:
+            if ticket[0]:
+                ticket[0].delete()
+            finalize()
+
+    data_object_and_associated_ticket = contextlib.contextmanager(_data_object_and_associated_ticket)
+
+    def test_redirect_in_data_object_put_and_get_with_tickets__issue_452(self):
+        for fname,size in {'large':40*MEBI,
+                           'small': 1*MEBI }.items():
+            content = b'1024123'*(size//7)
+            with self.data_object_and_associated_ticket(ticket_access='write') as data_ctx:
+                self._test_redirect_in_data_object_put_and_get__issue_452(content,data_ctx)
+            
+    def test_redirect_in_data_object_put_and_get_without_tickets__issue_452(self):
+        for fname,size in {'large':40*MEBI,
+                           'small': 1*MEBI  }.items():
+            content = b'1024123'*(size//7)
+            with self.data_object_and_associated_ticket(ticket_access='') as data_ctx:
+                self._test_redirect_in_data_object_put_and_get__issue_452(content,data_ctx)
+
+    def _test_redirect_in_data_object_put_and_get__issue_452(self, content, data_ctx):
```

Comment:
Please leave a comment explaining that the leading underscore is on purpose. Or better, add something to the name to indicate that this is a generic test that is not meant to be run directly (e.g. `do_test_redirect_in_data_object_put_and_get__issue_452`)